rust/src/measurements/make_private_expr/expr_count/mod.rs

```diff
@@ -0,0 +1,145 @@
+use crate::core::{ExprFunction, Measure, MetricSpace, PrivacyMap};
+use crate::domains::MarginPub;
+use crate::metrics::PartitionDistance;
+use crate::transformations::traits::UnboundedMetric;
+use crate::transformations::StableExpr;
+use crate::{
+    core::{Function, Measurement},
+    domains::ExprDomain,
+    error::Fallible,
+};
+
+use num::Zero;
+use polars::lazy::dsl::Expr;
+use polars_plan::dsl::AggExpr;
+
+/// Make a measurement that computes the count exactly under bounded-DP
+///
+/// | input_metric                              |
+/// | ----------------------------------------- |
+/// | `PartitionDistance<SymmetricDistance>`    |
+/// | `PartitionDistance<InsertDeleteDistance>` |
+///
+/// # Arguments
+/// * `input_domain` - ExprDomain
+/// * `input_metric` - valid selections shown in table above
+/// * `expr` - count expression
+pub fn make_expr_private_count<MI: 'static + UnboundedMetric, MO: 'static + Measure>(
+    input_domain: ExprDomain,
+    input_metric: PartitionDistance<MI>,
+    expr: Expr,
+) -> Fallible<Measurement<ExprDomain, Expr, PartitionDistance<MI>, MO>>
+where
+    MO::Distance: Zero,
+    Expr: StableExpr<PartitionDistance<MI>, PartitionDistance<MI>>,
+    (ExprDomain, PartitionDistance<MI>): MetricSpace,
+{
+    let Expr::Agg(AggExpr::Count(expr, include_nulls)) = expr else {
+        return fallible!(MakeMeasurement, "Expected Count expression");
+    };
+
+    if !include_nulls {
+        return fallible!(
+            MakeMeasurement,
+            "Public margin sizes assume nulls are included. Try using `len` instead."
+        );
+    }
+
+    let t_prior = expr
+        .clone()
+        .make_stable(input_domain.clone(), input_metric.clone())?;
+
+    let (middle_domain, middle_metric) = t_prior.output_space();
+
+    let by = middle_domain.context.grouping_columns()?;
+    let margin = middle_domain
+        .frame_domain
+        .margins
+        .get(&by)
+        .ok_or_else(|| err!(MakeMeasurement, "Unknown margin for {:?}", by))?;
+
+    if Some(MarginPub::Lengths) != margin.public_info {
+        return fallible!(
+            MakeMeasurement,
+            "Len is only private if size(s) of {:?} are public",
```

Comment:
Is "public" correct here? Is this related to the difference between length and size in Polars?